lib/netjockey/service.rb

```diff
@@ -62,16 +19,29 @@ def room(id)
       case res
       when Net::HTTPSuccess then
         data = JSON.parse(res.body)
-        room_info = RoomInfo.new(data['roomInfo']['name'],
-                                 data['roomInfo']['id'])
-        song_queue = data['queue']['playlist'].map do |song|
-          Song.new(song['duration'],
-                   song['title'], song['id'],
-                   song['thumbnailUrl'],
-                   song['uuid'])
-        end
-        Room.new(room_info, data['currentSongTime'], song_queue)
+        create_room(data)
+      when Net::HTTPNotFound then
+        raise 'room not found'
+      end
+    end
+
+    def create_room(room_hash)
+      room_info = RoomInfo.new(room_hash['roomInfo']['name'],
+                               room_hash['roomInfo']['id'])
+      Room.new(room_info,
+               room_hash['currentSongTime'],
+               create_songs(room_hash['queue']['playlist']))
+    end
+
+    def create_songs(song_queue_hash)
+      song_queue_hash.map do |song|
+        Song.new(song['duration'],
+                 song['title'], song['id'],
+                 song['thumbnailUrl'],
+                 song['uuid'])
       end
     end
+
+    private :create_room, :create_songs
```

Comment:
We usually separate public and private methods like this:
[https://github.com/Misenko/cloudkeeper/blob/master/lib/cloudkeeper/managers/image_list_manager.rb#L29](https://github.com/Misenko/cloudkeeper/blob/master/lib/cloudkeeper/managers/image_list_manager.rb#L29)
But it doesn't really matter.